tests/net/ieee802154/l2/src/ieee802154_test.c

```diff
@@ -444,36 +444,89 @@ static bool test_raw_packet_sending(void)
 	return result;
 }
 
-static bool test_ack_reply(struct ieee802154_pkt_test *t)
+static bool test_recv_and_ack_reply(struct ieee802154_pkt_test *t)
 {
+	/* Incoming IEEE 802.15.4 packet with payload header compression. */
 	static uint8_t data_pkt[] = {
-		0x61, 0xdc, 0x16, 0xcd, 0xab, 0xc2, 0xa3, 0x9e, 0x00, 0x00, 0x4b,
-		0x12, 0x00, 0x26, 0x18, 0x32, 0x00, 0x00, 0x4b, 0x12, 0x00, 0x7b,
-		0x00, 0x3a, 0x20, 0x01, 0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00,
-		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x20, 0x01, 0x0d, 0xb8,
-		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-		0x02, 0x87, 0x00, 0x8b, 0x00, 0x00, 0x00, 0x00, 0x00, 0x20, 0x01,
-		0x0d, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff,
-		0x16, 0xf0, 0x02, 0xff, 0x16, 0xf0, 0x12, 0xff, 0x16, 0xf0, 0x32,
-		0xff, 0x16, 0xf0, 0x00, 0xff, 0x16, 0xf0, 0x00, 0xff, 0x16
+		0x61, 0xd8, 0x16, 0xcd, 0xab, 0xff, 0xff, 0x00, 0x12, 0x4b, 0x00, 0x1c, 0xa1, 0xac,
+		0xe5, 0x7b, 0x39, 0x3a, 0x02, 0x01, 0xff, 0x4b, 0x12, 0x00, 0x87, 0x00, 0x0d, 0xa4,
+		0x00, 0x00, 0x00, 0x00, 0xfe, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x70, 0x14,
+		0xa6, 0x1c, 0x00, 0x4b, 0x12, 0x00, 0x01, 0x02, 0xe5, 0xac, 0xa1, 0x1c, 0x00, 0x4b,
+		0x12, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	};
+	/* Expected uncompressed IPv6 payload. */
+	static uint8_t expected_rx_pkt[] = {
+		0x60, 0x00, 0x00, 0x00, 0x00, 0x28, 0x3a, 0xff, 0xfe, 0x80, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0xe7, 0xac, 0xa1, 0x1c, 0x00, 0x4b, 0x12, 0x00, 0xff, 0x02, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0xff, 0x4b, 0x12, 0x00, 0x87, 0x00,
+		0x0d, 0xa4, 0x00, 0x00, 0x00, 0x00, 0xfe, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x70, 0x14, 0xa6, 0x1c, 0x00, 0x4b, 0x12, 0x00, 0x01, 0x02, 0xe5, 0xac, 0xa1, 0x1c,
+		0x00, 0x4b, 0x12, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
```

Comment:
This is the way I have done the testing too in other parts of the net test and that is fine. I am just thinking the poor soul who might need to change something in the test data. So just in case, could you add some more comments to the data in order to "partition" it in more helpful way.
For example something like what is done here https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/net/ipv6/src/main.c#L48

Reply:
Of course, that makes a lot of sense. Fixed.